extide/gradle/src/org/netbeans/modules/gradle/loaders/LegacyProjectLoader.java

```diff
@@ -472,8 +474,20 @@ private static NbProjectInfo retrieveProjectInfo(NbGradleProjectImpl projectImpl
             offline.addFlag(GradleCommandLine.Flag.OFFLINE);
         }
 
+        File javaHome = null;
+        GradleJavaPlatformProvider jvmProvider = projectImpl.getLookup().lookup(GradleJavaPlatformProvider.class);
+        try {
+            if (jvmProvider != null) {
+                javaHome = jvmProvider.getJavaHome();
+                LOG.log(Level.FINE, "Using JAVA_HOME=''{0}'' for project info load for: {1}", new Object[]{javaHome, projectImpl});
+            }
+        } catch (FileNotFoundException ex) {
+            LOG.log(Level.WARNING, "JAVA_HOME for project " + projectImpl + " not found.", ex);
+        }
+
         if (goOnline == GoOnline.NEVER || goOnline == GoOnline.ON_DEMAND) {
             BuildActionExecuter<NbProjectInfo> action = createInfoAction(pconn, offline, token, pl);
+            action.setJavaHome(javaHome);
```

Comment:
what if javaHome variable is null here?

Reply:
From: https://docs.gradle.org/current/javadoc/org/gradle/tooling/ConfigurableLauncher.html#setJavaHome-java.io.File-

> If not configured or null is passed, then the sensible default will be used.